api/src/main/java/io/minio/MinioClient.java

```diff
@@ -3359,6 +3359,65 @@ public void makeBucket(String bucketName, String region, boolean objectLock)
     response.body().close();
   }
 
+  /**
+   * Creates a bucket makeBucketArgs.
+   *
+   * <pre>Example:{@code
+   * minioClient.makeBucket(MakeBucketArgs makebucketArgs);
+   * }</pre>
+   *
+   * @param makebucketArgs Object with bucket name, region and lock functionality
+   * @throws ErrorResponseException thrown to indicate S3 service returned an error response.
+   * @throws IllegalArgumentException throws to indicate invalid argument passed.
+   * @throws InsufficientDataException thrown to indicate not enough data available in InputStream.
+   * @throws InternalException thrown to indicate internal library error.
+   * @throws InvalidBucketNameException thrown to indicate invalid bucket name passed.
+   * @throws InvalidKeyException thrown to indicate missing of HMAC SHA-256 library.
+   * @throws InvalidResponseException thrown to indicate S3 service returned invalid or no error
+   *     response.
+   * @throws IOException thrown to indicate I/O error on S3 operation.
+   * @throws NoSuchAlgorithmException thrown to indicate missing of MD5 or SHA-256 digest library.
+   * @throws RegionConflictException thrown to indicate passed region conflict with default region.
+   * @throws XmlParserException thrown to indicate XML parsing error.
+   */
+  public void makeBucket(MakeBucketArgs makebucketArgs)
+      throws InvalidBucketNameException, RegionConflictException, InsufficientDataException,
+          InternalException, InvalidResponseException, InvalidKeyException,
+          NoSuchAlgorithmException, XmlParserException, ErrorResponseException, IOException {
+
+    checkBucketName(makebucketArgs.bucketName());
```

Comment:
0. Just have `public void makeBucket(MakeBucketArgs args)` and check `args` is not `null`.
1. All such validations need to go into `MakeBucketArgs.Builder` and throw `IllegalArgumentException` on error.
2. All existing `makeBucket()` should use this `makeBucket()` and they can be deprecated.
3. As all existing `makeBucket()` are deprecated, we would need to move examples/functional tests to use this `makeBucket()`